src/sgame/sg_bot_util.cpp

```diff
@@ -1459,6 +1486,50 @@ bool PlayersBehindBotInSpawnQueue( gentity_t *self )
 	}
 }
 
+// Initializes numUpgrades and numWeapons arrays with team's current equipment
+// pre-condition:
+// * self points to a player which is inside a playable team
+// * numUpgrades is an array of at least maxUpgrades unsigned integers
+// * numUpgrades memory is initialized
+// * numWeapons is an array of at least maxWeapons unsigned integers
+// * numWeapons memory is initialized
+// * self's team have less than MAX_CLIENTS players
+void TeamEquipment( gentity_t *self, unsigned int* numUpgrades, size_t maxUpgrades, unsigned int* numWeapons, size_t maxWeapons )
+{
+	ASSERT( self );
+	ASSERT( numUpgrades );
+	ASSERT( numWeapons );
+	const team_t team = static_cast<team_t>( self->client->pers.team );
+	unsigned int alliesNumbers[MAX_CLIENTS];
+	memset( alliesNumbers, 0, sizeof( alliesNumbers ) );
+	unsigned int numAllies = FindAllies( alliesNumbers, MAX_CLIENTS, team );
+
+	for ( unsigned int i = 0; i < numAllies; ++i )
+	{
+		gentity_t *ally = &g_entities[alliesNumbers[i]];
+		if ( ally == self )
+		{
+			continue;
+		}
+		++numWeapons[ally->client->ps.stats[STAT_WEAPON]];
+		if ( team == TEAM_HUMANS )
+		{
+			// for all possible upgrades, check if current player already have it.
+			// this is very fragile, at best, since it will break if TEAM_ALIENS becomes
+			// able to have upgrades. Class is not considered, neither.
+			// last but not least, it's playing with bits.
+			int * stat = &ally->client->ps.stats[STAT_ITEMS];
+			for ( int up = UP_NONE + 1; up < UP_NUM_UPGRADES; ++up )
+			{
+				if ( *stat & ( 1 << up ) )
```

Comment:
If the `*stat` is constant, why check it on each loop? Also, I don't see what it purpose is, which means it could use a better name (but maybe it shouldn't exist at all?).